### ISSUE TITLE
Tightly coupled network layer and view model

### ISSUE THREAD

**Issue by jasonnam** [Arunjos/SurveyTaker#2]
Currently network layer is tightly coupled with survey view model. This makes network layer hard to be tested and reused.

**Comment by Arunjos** [comment]
> Currently network layer is tightly coupled with survey view model. This makes network layer hard to be tested and reused.

Yes right when we extent the application to next level it will difficult maintain. Now I resolved the issue and merged the pull request to master please verify.
